cmd/ingest/main.go

```diff
@@ -90,117 +105,158 @@ func main() {
 		}).Fatal("couldn't create SQL tables")
 	}
 
-	// Initialize ZMQ
-	ctx, err := zmq.NewContext()
+	//Initialize RPC connection with full node zcashd
+	rpcClient, err := frontend.NewZRPCFromConf(opts.zcashConfPath)
 	if err != nil {
 		log.WithFields(logrus.Fields{
 			"error": err,
-		}).Fatal("couldn't create zmq context")
+		}).Warn("zcash.conf failed, will try empty credentials for rpc")
+
+		//Default to testnet, but user MUST specify rpcuser and rpcpassword in zcash.conf; no default
+		rpcClient, err = frontend.NewZRPCFromCreds(opts.rpcHost+":"+opts.rpcPort, opts.rpcLogin, opts.rpcPass)
+
+		if err != nil {
+			log.WithFields(logrus.Fields{
+				"error": err,
+			}).Fatal("couldn't start rpc connection")
+		}
 	}
-	defer ctx.Term()
 
-	// WARNING: The Socket is not thread safe. This means that you cannot
-	// access the same Socket from different goroutines without using something
-	// like a mutex.
-	sock, err := ctx.NewSocket(zmq.SUB)
+	ctx := context.Background()
+	height, err := storage.GetCurrentHeight(ctx, db)
 	if err != nil {
 		log.WithFields(logrus.Fields{
 			"error": err,
-		}).Fatal("couldn't create zmq context socket")
-	}
+  	}).Warn("unable to get current height from local db storage")
 
-	err = sock.SetSubscribe(opts.zmqTopic)
-	if err != nil {
-		log.WithFields(logrus.Fields{
-			"error":  err,
-			"stream": opts.zmqTopic,
-		}).Fatal("couldn't subscribe to stream")
 	}
 
-	connString := fmt.Sprintf("tcp://%s", opts.zmqAddr)
-
-	err = sock.Connect(connString)
-	if err != nil {
+	//ingest from Sapling testnet height
```

Comment:
the comment here reads "testnet" but this is the "mainnet" start height. To prevent issues like this, in general, we may want to provide a more explicit way to handle `mainnet` v. `testnet` configuration. Perhaps, we would read that information from the `zcash.conf` file and choose a start height, accordingly. Rather than specifying it as a command-line option and then validating the input.